azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -299,20 +301,23 @@ protected Azure getAzureClientByAuthType() throws AzureAuthFailureException {
     }
 
     protected AuthType getAuthTypeEnum() {
```

Comment:
wrap the logic of converting authType string to azure cred wrapper in a helper method.